astropy/modeling/functional_models.py

```diff
@@ -2508,3 +2508,138 @@ def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+        import matplotlib.pyplot as plt
+
+        plt.figure()
+        rt_list = [1, 2, 5, 10, 20]
+        for rt in rt_list:
+            r = np.linspace(0.1, rt, 100)
+
+            mod = KingProjectedAnalytic1D(amplitude = 1, r_core = 1., r_tide = rt)
+            sig = mod(r)
+
+
+            plt.loglog(r, sig/sig[0], label='c ~ {:0.2f}'.format(mod.concentration))
+
+        plt.xlabel("r")
+        plt.ylabel(r"$\\sigma/\\sigma_0$")
+        plt.legend()
+        plt.show()
+
+    References
+    ----------
+    .. [1] http://articles.adsabs.harvard.edu/pdf/1962AJ.....67..471K
+    """
+
+    amplitude = Parameter(default=1, bounds=(FLOAT_EPSILON, None))
+    r_core = Parameter(default=1, bounds=(FLOAT_EPSILON, None))
+    r_tide = Parameter(default=2, bounds=(FLOAT_EPSILON, None))
+
+    @property
+    def concentration(self):
+        """Concentration parameter of the king model"""
+        return np.log10(np.abs(self.r_tide/self.r_core))
+
+    @staticmethod
+    def evaluate(x, amplitude, r_core, r_tide):
+        """
+        Analytic King model function.
+        """
+
+        result = amplitude * r_core ** 2 * (1/np.sqrt(x ** 2 + r_core ** 2) -
+                                          1/np.sqrt(r_tide ** 2 + r_core ** 2)) ** 2
+
+        # Set invalid r values to 0
+        bounds = (x >= r_tide) | (x<0)
+        result[bounds] = result[bounds] * 0.
+
+        return result
+
+    @staticmethod
+    def fit_deriv(x, amplitude, r_core, r_tide):
+        """
+        Analytic King model function derivatives.
+        """
+        d_amplitude = r_core ** 2 * (1/np.sqrt(x ** 2 + r_core ** 2) -
+                                     1/np.sqrt(r_tide ** 2 + r_core ** 2)) ** 2
+
+        d_r_core = 2 * amplitude * r_core ** 2 * (r_core/(r_core ** 2 + r_tide ** 2) ** (3/2) -
+                                                  r_core/(r_core ** 2 + x ** 2) ** (3/2)) * \
+                   (1./np.sqrt(r_core ** 2 + x ** 2) - 1./np.sqrt(r_core ** 2 + r_tide ** 2)) + \
+                   2 * amplitude * r_core * (1./np.sqrt(r_core ** 2 + x ** 2) -
+                                             1./np.sqrt(r_core ** 2 + r_tide ** 2)) ** 2
+
+        d_r_tide = (2 * amplitude * r_core ** 2 * r_tide *
+                    (1./np.sqrt(r_core ** 2 + x ** 2) -
+                     1./np.sqrt(r_core ** 2 + r_tide ** 2)))/(r_core ** 2 + r_tide ** 2) ** (3/2)
+
+        # Set invalid r values to 0
+        bounds = (x >= r_tide) | (x < 0)
+        d_amplitude[bounds] = d_amplitude[bounds]*0
+        d_r_core[bounds] = d_r_core[bounds]*0
+        d_r_tide[bounds] = d_r_tide[bounds]*0
+
+        return [d_amplitude, d_r_core, d_r_tide]
+
+    @property
+    def bounding_box(self):
+        """
+        Tuple defining the default ``bounding_box`` limits.
+
+        The model is not defined for r > r_tide.
+
+        ``(r_low, r_high)``
+        """
+
+        return (0 * self.r_tide, 1 * self.r_tide)
```

Comment:
Can this return simply `(0, self.r_tide)`?

Reply:
It did originally but then I had problems with the 3.6 tests failing  and I think it was because of that. Also it would then not have units for the first limit and this caused problems with  the pytests.

Reply:
Oh, yes, good point about the units!